File: backend/app/schemas/openapi_knowledge.py
"""OpenAPI 知识库接口 Schemas。"""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class _BaseSchema(BaseModel):
    model_config = {"from_attributes": True}


class OpenAPIDocumentResponse(_BaseSchema):
    id: int
    filename: str
    status: str
    endpoint_count: int
    chunk_count: int
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime


class OpenAPIDocumentListResponse(BaseModel):
    items: list[OpenAPIDocumentResponse]
    total: int
    skip: int
    limit: int


class OpenAPIDocumentUploadResponse(BaseModel):
    document_id: int
    filename: str
    status: str


class OpenAPIEndpointResponse(_BaseSchema):
    id: int
    document_id: int
    chunk_id: str
    path: str
    method: str
    summary: str | None = None
    description: str | None = None
    tags: list[str] | None = None
    operation_id: str | None = None
    content: str
    created_at: datetime
    updated_at: datetime


class OpenAPIEndpointListResponse(BaseModel):
    items: list[OpenAPIEndpointResponse]
    total: int
    skip: int
    limit: int


class OpenAPISearchResultResponse(BaseModel):
    id: int
    document_id: int
    chunk_id: str
    path: str
    method: str
    summary: str | None = None
    description: str | None = None
    tags: list[str] | None = None
    operation_id: str | None = None
    content: str
    similarity_score: float = Field(..., ge=0.0, le=1.0)


class OpenAPISearchResponse(BaseModel):
    items: list[OpenAPISearchResultResponse]
    total: int

File: frontend/src/types/openapiKnowledge.ts
export type OpenAPIDocumentStatus = 'pending' | 'processing' | 'completed' | 'failed'

export interface OpenAPIDocument {
  id: number
  filename: string
  status: OpenAPIDocumentStatus
  endpoint_count: number
  chunk_count: number
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface OpenAPIDocumentUploadResponse {
  document_id: number
  filename: string
  status: OpenAPIDocumentStatus
}

export interface OpenAPIDocumentListResponse {
  items: OpenAPIDocument[]
  total: number
  skip: number
  limit: number
}

export interface OpenAPIEndpoint {
  id: number
  document_id: number
  chunk_id: string
  path: string
  method: string
  summary: string | null
  description: string | null
  tags: string[] | null
  operation_id: string | null
  content: string
  created_at: string
  updated_at: string
}

export interface OpenAPIEndpointListResponse {
  items: OpenAPIEndpoint[]
  total: number
  skip: number
  limit: number
}

export interface OpenAPISearchResult {
  id: number
  document_id: number
  chunk_id: string
  path: string
  method: string
  summary: string | null
  description: string | null
  tags: string[] | null
  operation_id: string | null
  content: string
  similarity_score: number
}

export interface OpenAPISearchResponse {
  items: OpenAPISearchResult[]
  total: number
}

export interface ListOpenAPIDocumentsParams {
  skip?: number
  limit?: number
  status?: OpenAPIDocumentStatus
}

export interface ListOpenAPIEndpointsParams {
  skip?: number
  limit?: number
  document_id?: number
  method?: string
  tag?: string
}

export interface SearchOpenAPIKnowledgeParams {
  q: string
  top_k?: number
  method?: string
  document_id?: number
}
